pyspeckit/spectrum/units.py

```diff
@@ -417,6 +423,11 @@ def __new__(self, xarr, unit="Hz", frame='rest', frame_offset=0.0,
         else:
             subarr.velocity_convention = velocity_convention
 
+        subarr._equivalencies = equivalencies
+        if center_frequency:
+            subarr.center_frequency, subarr._equivalencies  = self.find_equivalencies(velocity_convention, center_frequency, center_frequency_unit, equivalencies)
+        subarr.center_frequency_unit = center_frequency_unit
```

Comment:
The `center_frequency_unit` should now just be `subarr.center_frequency.unit` - we don't need to store it any more